src/bsoncxx/exception/exception.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bsoncxx/config/prelude.hpp>
+
+#include <bsoncxx/exception/exception.hpp>
+
+namespace bsoncxx {
+BSONCXX_INLINE_NAMESPACE_BEGIN
+namespace exception {
+
+exception::exception(std::error_code ec, std::string what_arg)
+    : std::system_error(ec, what_arg), _ec(ec), _what(what_arg) {
+}
+
+const std::error_code& exception::code() const {
```

Comment:
Do you need this, or is it inherited from system_error? (And if not, I should probably include it in mongocxx::exception.)